### ISSUE TITLE
[Bug]: Inventory and position Rollback

### ISSUE THREAD

**Issue by MRGuyyy** [FTBTeam/FTB-Modpack-Issues#1350]
### Modpack

FTB Academy and University 1.12

### Modpack version

1.4.0 and 1.2.1

### Log Files

I have no log available, as the PC was formatted recently

### Describe the bug

In the FTB Academy I played the modpack normally for about a week and three days, the next day I opened my world and saw that all my inventory and my world position had been rolled back to when I had iron armor and stone tools, I lost my backpack, some Tinkers tools and other tools like the Resonating Wand, Jetpack and etc.

In the case of FTB University it was exactly the same, but it took me a while to realize that I had stellar tools that were rolled back to end steel and I thought they were similar.

### Steps to reproduce

1. Log into the game create a world and play as normal
2. Every time you leave the world, copy the world folder to a USB stick, thus making a backup
3. I played for a week and some two or three days
4. I opened the world as usual
5. I realized that I was somewhere else
6. I noticed that my inventory had rolled back

### Expected behaviour

Just enter the world and continue playing as normal

### Screenshots

I have no screenshots of what happened

### Additional information

The two modpacks are from version 1.12, the Academy being version 1.4.0 and the University being 1.2.1 both downloaded from the FTB App

I don't know how or why this happened, and in two modpacks, unfortunately I was afraid to start other worlds, because I believe the same will happen again

I don't know if it has anything to do with this, but I used to leave a USB stick plugged in my PC all the time containing backups of all the modpacks worlds I had including these two, every time I quit the game I backed up the world to USB. or if it was because I removed the backup mod from both modpacks (on the University after it happened and on the Academy before it happened)

**Comment by Raidobw2** [comment]
That's an odd case, the game usually can't roll itself back. I've played both modpacks before, in singleplayer and on server with friends and never had these issues, so I think possibly something else than the game is interacting with modpack files.

Do you have an antivirus maybe?

Also, in singleplayer the player's inventory and position are stored within the world, in the `level.dat` file. Say your world is called "world" then the file would be at `world/level.dat`. If there's anything you want to check or keep intact, it's that file.

The `world/region` folder would instead contain all your base in the Overworld only (not the Nether or End).

I'm not really sure how exactly you copied-pasted the files; let me know if the information above might help shining some light on the incident.
You can also check in the `backups` folder (that folder is next to the `saves` folder) if you have any backups. These would contain the important `level.dat` files with your inventory most likely.

Also, if you can manage to reproduce this and post your game logs, we'll know exactly what happens I think. When the world files are locked and the world cannot be properly saved, it's written in the game log. This way we'll know if something is preventing your world to save properly.

There's ways to see which program is preventing it from saving (Process Explorer), but I won't get into this yet, I'll wait for your feedback first and thanks.

**Comment by MRGuyyy** [comment]
Yes, I have Avast as antivirus, and also when I went to look for something in the backup folder to try to recover my items, it was only with the data from the current inventory which had already been rolled back, there was simply nothing from previous days

**Comment by Raidobw2** [comment]
I see, thanks for the information. If you play for a while again, make sure to post your game logs here, that'll help a lot. I know of some problems with Minecraft when Avast is installed, but they mostly affect Java itself. It's possible it could be messing with the save files, if you want to test it you would have to turn off Avast when you play and observe if anything weird happens or the world isn't being saved properly.

That would be the only track we have now, that would be my suggestion. Let me know how it goes when you play any pack again

**Comment by MRGuyyy** [comment]
After replaying the modpack I realized that the problem had nothing to do with making backups to the USB stick, but rather excessive crashes.

I remember that at the time I played there was a warning in the top right corner of the screen that "something has gone wrong" as if it had predicted a crash, and I also remember that when I had this bug, the game had crashed several times, about five I think.

Replaying the pack, I was more careful, didn't have any crashes and managed to finish the modpack.
